onnx/reference/ops/op_quantize_linear.py

```diff
@@ -63,25 +63,52 @@ def common_run(
         axis: int = 1,
         saturate: bool = True,
     ) -> tuple[np.ndarray]:
-        if len(y_scale.shape) > 1:
-            raise RuntimeError("Input 2 must be a vector or a number.")
-        if len(y_scale.shape) > 0 and y_scale.size == 1:
-            y_scale = y_scale[0]
-        if len(y_scale.shape) > 0:
-            new_shape = [1 for s in x.shape]
+        if len(y_scale.shape) == 0 or y_scale.size == 1:  # per-tensor
+            if len(y_scale.shape) > 0:
+                y_scale = y_scale[0]
+            x = x / y_scale
+            new_shape = x.shape  # unused
+        elif len(y_scale.shape) == 1 and y_scale.size > 1:  # per-axis
```

Comment:
What about a 1D tensor quantized by blocks?

Reply:
Blocking of 1D tensor is not supported. I don't think there's a use case for this, and in any case the same result could be achieved by expanding the input dims, or using reshape + per axis.
If we would want to support blocking of 1D tensors, we won't be able to determine the type of quantization (per-axis or blocked) from the scale's shape, meaning adding an attribute to the Q/DQ operators will be required (possible, but IMO not desirable).

In any case, I should have noted this in the operator's definition. I'm keeping this thread open as a reminder to do so.

Reply:
If it is not supported, it should be mentioned I think and tell if shape inference would fail in that case.

Reply:
This is mentioned in revised operator definition.
For an input tensor with shape (n, ), quantization scale shape can be either (1,) - per-tensor Q, or (n, ) - per-axis Q.
Any other option will fail.